pkg/controller/handler.go

```diff
@@ -227,6 +229,31 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 				ResetFinalizers(mutatedWf)
 			}
 		}
+
+		// ExecutionNotFound error is returned when flyteadmin is missing the workflow. This is not
+		// a valid state unless we are experiencing a race condition where the workflow has not yet
+		// been inserted into the db (ie. workflow phase is WorkflowPhaseReady).
+		if err != nil {
+			e, ok := err.(*errors.WorkflowErrorWithCause)
+			if ok && eventsErr.IsNotFound(e.Cause()) && w.GetExecutionStatus().GetPhase() != v1alpha1.WorkflowPhaseReady {
+				t.Stop()
+				logger.Errorf(ctx, "Failed to process workflow, failing: %s", e.Cause())
+
+				// We set the workflow status to failing to abort any active tasks in the next round.
+				mutableW := w.DeepCopy()
+				mutableW.Status.UpdatePhase(v1alpha1.WorkflowPhaseFailing, "Workflow execution is missing in flyteadmin, aborting", &core.ExecutionError{
```

Comment:
I forgot how we handle WorkflowPhaseFailing, do we try to send any events while handling that phase? if we do, those will also fail... and we will get stuck in that phase...

Reply:
You are correct, the WorkflowPhaseFailing does attempt to transition phase and send an event to admin. I added a check (and comment) in pkg/controller/workflow/executor.go to ignore the ExecutionNotFound error for the Failing to Failed transition. I think this makes sense because if flyteadmin is missing the execution we don't need to add that error to the list of reasons the workflow is failing, we can just gracefully fail. The error is still logged in multiple locations.